src/alfasim_sdk/alfacase/generate_schema.py

```diff
@@ -0,0 +1,389 @@
+        )
+        raise TypeError(msg)
+    attribute_name = _get_attr_name(key, value)
+    attribute_value = _get_attr_value(
+        value.type, block_indentation=indentation, key=key
+    )
+
+    return indentation + f"{attribute_name}: {attribute_value}" + ","
+
+
+def generate_alfacase_schema(class_: type) -> str:
+    """
+    Return a schema for the given `class_` that is decorated with @attr and has type hints
+    """
+    lines = [f"{obtain_schema_name(class_)} = Map(", f"{INDENTANTION}{{"]
+    lines.extend(
+        _get_attribute_schema(key, value)
+        for key, value in attr.fields_dict(class_).items()
+        if key not in IGNORED_PROPERTIES
+    )
+    lines.extend([f"{INDENTANTION}}}", ")", ""])
+
+    return "\n".join(lines)
+
+
+def _is_from_typing_module(type_: type) -> bool:
+    """Helper method to verify if the given `type_` is instance of List/Dict/Union"""
+    return is_list(type_) or is_dict(type_) or typing_inspect.is_union_type(type_)
+
+
+def _obtain_referred_type(type_) -> List[type]:
+    """
+    Obtain the type referred by the type hint.
+
+    Example.:
+        class A()
+            first: List[str]
+            second: List[Foo]
+            third: Union[Foo, Bar]
+            fourth: List[Union[Foo, Bar]]
+            fifth: Union[str, int]
+
+    Running `_ObtainReferredType` on each field will return:
+        first => [str]
+        second => [Foo]
+        third => [Foo, Bar]
+        fourth => [Foo, Bar]
+        fifth => [str, int]
+
+    If type_ is not either a List or Union, an TypeError exception will be raised.
+    """
+
+    if is_list(type_):
+        type_ = typing_inspect.get_args(type_)[0]
+        return (
+            _obtain_referred_type(type_) if _is_from_typing_module(type_) else [type_]
+        )
+
+    if typing_inspect.is_union_type(type_):
+        return [i for i in type_.__args__ if i is not None.__class__]
+
+    if is_dict(type_):
+        return type_.__args__[1]
+
+    raise TypeError(f"type_ must be a List or Union referring other types, got {type_}")
+
+
+def _get_classes(class_: type) -> Set[type]:
+    """
+    Helper function to return a set of attr classes that needs schema
+    """
+
+    def needs_schema(type_):
+        """Helper function to ensure that the given attribute (type_) needs a strict yaml schema"""
+        return (
+            is_attrs(type_)
+            or _is_from_typing_module(type_)
+            and is_attrs(_obtain_referred_type(type_))
+        )
+
+    def get_attr_class_type(type_):
+        """Helper function to retrieve the attr type being referred by the type hint"""
+        return type_ if is_attrs(type_) else _obtain_referred_type(type_)
+
+    classes = []
+    for key, value in attr.fields_dict(class_).items():
+        if needs_schema(value.type) and not key in IGNORED_PROPERTIES:
+            classes.append(get_attr_class_type(value.type))
+
+    return set(sorted(flatten(classes), key=operator.attrgetter("__name__")))
```

Comment:
Why `sorted` is required? There is no point in sorting the elements in a set.
```suggestion
    return set(flatten(classes))
```